domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgress.kt

```diff
@@ -90,4 +110,22 @@ internal class ExplorationProgress {
     /** The controller is in the process of submitting an answer. */
     SUBMITTING_ANSWER
   }
+
+  /** Different states in which checkpoint saving exploration progress can exist. */
+  enum class CheckpointState {
+    /**
+     *  Progress made in the exploration is saved and the size of the checkpoint database has
+     * not exceeded limit.
+     */
+    SAVED_CHECKPOINT_DATABASE_NOT_EXCEEDED_LIMIT,
+
+    /**
+     * Progress made in the exploration is saved and the size of the checkpoint database has
+     * exceeded limit.
+     */
+    SAVED_CHECKPOINT_DATABASE_EXCEEDED_LIMIT,
+
+    /** Progress made in the exploration is not saved. */
+    UNSAVED
+  }
 }
```

Comment:
This enum is very similar to the one we are using in ExplorationCheckpointController. 
We can create a new file with just this enum and use that in domain and app layer.

We can also create a proto something like this 
```
// structure that holds the current state of the checkpoint for any exploration.
message ExplorationCheckpointState {
  // Boolean that indicates whether the latest progress made in the exploration is saved.
  bool isProgressSaved = 1;

  // Boolean that indicates if the checkpoint database can store more progress or not.
  bool databaseCanStoreProgress = 2;
}

```
WDYT?